src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -1,274 +1,22 @@
-    get dirty() {
-      return formatDateString(props.value) !== this.value;
-    },
-    id: props.id,
-    name: props.name,
-    get value() {
-      if (inputRef.current) {
-        return formatDateString(inputRef.current.value);
-      } else if (componentRef.current) {
-        return formatDateString(componentRef.current.value);
-      }
-      return undefined;
-    },
-  }));
-
-  useEffect(() => {
-    if (!initialValue) {
-      if (ref) {
-        props.onChange({ target: ref.current });
-      } else {
-        props.onChange(currentValue);
-      }
-    } else {
-      setInitialValue(false);
-    }
-  }, [currentValue]);
-
-  useEffect(() => {
-    setCurrentValue(parseValue(props.value));
-  }, [props.value]);
-
-  const handleOnChange = (event) => {
-    if (inputRef.current) {
-      const isInputValid = inputRef.current.validity.valid;
-      const newDate = formatDateString(event.target.value);
-      setIsValid(isValueValid(newDate, props.errorText, isInputValid));
-    }
-
-    setCurrentValue(event.target.value);
-  };
-
-  const errorText = () => {
-    if (!isValid && !isValidInput(props.value)) {
-      return `"${props.value}" is an invalid date`;
-    }
-
-    return props.errorText;
-  };
-
-  const wrapperRef = useRef(null);
-  const handleDropdownClose = () => {
-    setExpanded(false);
-    setIsEditing(false);
-  };
-  useDropdownClose(wrapperRef, expanded, handleDropdownClose);
-
-  const openCalendar = () => {
-    if (!props.readOnly) {
-      setExpanded(!expanded);
-      setIsEditing(!isEditing);
-    }
-  };
-
-  function onInputClick(event) {
-    event.preventDefault();
-    if (!props.readOnly) {
-      setExpanded(true);
-      setIsEditing(true);
-    }
-  }
-
-  function onInputKeyDown(event) {
-    if (event.key === 'Enter' || event.key === 'Space') {
-      event.preventDefault();
-      if (!props.readOnly) {
-        setShouldFocusInput(true);
-        setExpanded(!expanded);
-        setIsEditing(!isEditing);
-      }
-    }
-  }
-
-  function onKeyDown(event) {
-    if (event.key === 'Escape') {
-      setExpanded(false);
-      setIsEditing(false);
-    }
-  }
-
-  function calendarIcon() {
-    if (props.readOnly) {
-      return (
-        <Icon
-          icon={calendarSvg}
-          title={props.label}
-          label={`${props.label} calendar`}
-        />
-      );
-    }
-
-    return (
-      <IconButton
-        onPress={openCalendar}
-        icon={calendarSvg}
-        title={props.label}
-        label={`${props.label} calendar button`}
-      />
-    );
-  }
-
-  const sharedProps = {
-    className: dateStyles['date-input'],
-    inputClassName: props.errorText ? dateStyles['input-invalid'] : dateStyles.input,
-    icon: calendarIcon(),
-    label: props.label,
-    required: props.required,
-    onClick: onInputClick,
-    onKeyDown: onInputKeyDown,
-    errorText: props.errorText,
+  const classNames = {
+    ...(props.className ? { input: props.className } : {}),
   };
```

Comment:
I don't think we need the extraneous spreadage do we?

```suggestion
  const classNames = props.className ? { input: props.className } : {};
```